src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -455,20 +455,35 @@ private CompatibilityData GetCompatibilityData(RestoreTargetGraph graph, Library
         {
             // Use data from the current lock file if it exists.
             LockFileTargetLibrary targetLibrary = null;
-            var target = _lockFile.Targets.FirstOrDefault(t => Equals(t.TargetFramework, graph.Framework) && string.Equals(t.RuntimeIdentifier, graph.RuntimeIdentifier, StringComparison.Ordinal));
-            if (target != null)
+
+            for (int i = 0; i < _lockFile.Targets.Count; ++i)
             {
-                targetLibrary = target.Libraries
-                    .FirstOrDefault(t => t.Name.Equals(libraryId.Name, StringComparison.OrdinalIgnoreCase) && t.Version.Equals(libraryId.Version));
+                var target = _lockFile.Targets[i];
+                if (Equals(target.TargetFramework, graph.Framework) && string.Equals(target.RuntimeIdentifier, graph.RuntimeIdentifier, StringComparison.Ordinal))
+                {
+                    for (int j = 0; j < target.Libraries.Count; ++j)
+                    {
+                        var library = target.Libraries[j];
```

Comment:
The old code would iterate `_lockFile.Targets` and stop when the TFM & RID match, then will look though `target.Libraries`. But if it couldn't find the patching package/project name in libraries, it wouldn't resume searching though `_lockFile.Targets`.

While NuGet's data model doesn't prevent it, there should never be a case of a TFM & RID combo being duplicated in `_lockFile.Targets`.

Therefore, this PR's change, where it will resume iterating the rest of the targets, is doing unnecessary CPU work. Having said that, even for MAUI projects, the `_lockFile.Targets` list isn't huge. But this PR is removing an early exit optimization.

Reply:
I could be missing something, but I believe this maintains the early exit. The `break` on line 474 will break out of the outermost loop on the first TMF & RID match.

Reply:
oops, yes, I missed that.